features:

pdziugas opened the pull request:
- data is now fetched from db.json using http and json servers
- data is displayed in carousel and menu itself

pdziugas committed:
features:
- data is now fetched from db.json using http and json servers
- data is displayed in carousel and menu itself

LukasTyla commented on index.html:
```diff
@@ -47,90 +64,15 @@
-        <div class="control next"></div>
-      </div>
+      <div class="slides"></div>
+      <div class="controls"></div>
```
Couldn't you have left the controls in the DOM itself? 🤔

LukasTyla commented on js/carousel.js:
```diff
@@ -1,33 +1,71 @@
+  document.querySelector(".controls").innerHTML = "";
+
+  // display controls
+  document.querySelector(".controls").innerHTML += `
```
Same suggestion here, you can put these elements in the DOM and just add/remove event listeners when/if needed. 😉

LukasTyla commented on js/displayProducts.js:
```diff
@@ -0,0 +1,22 @@
+const displayProducts = (productData) => {
+  document.querySelector(".container").innerHTML = "";
+  productData.forEach((product) => {
+    document.querySelector(".container").innerHTML += `
```
Same recommendation for minimising DOM operations applies here as well. 😉

LukasTyla commented on package.json:
```diff
@@ -4,7 +4,10 @@
+    "compile:sass": "node-sass sass/main.scss css/style.css -w",
+    "json:server": "json-server api/db.json --routes api/routes.json",
+    "http:server": "http-server --proxy http://localhost:3000 -c",
+    "dev": "concurrently --kill-others \"npm run compile:sass\" \"npm run json:server\" \"npm run http:server\""
```
👍 
Just an observation, but if you already have `node-sass` installed in project level, then maybe it would be best to also add json-server and `http-server` as dependencies and increase the project portability? 😃

LukasTyla commented on api/routes.json:
```diff
@@ -0,0 +1,6 @@
+{
```
Do you really need any of these configurations? 🤔
I reckon that all your needs are covered by implicitly created api from db.json file.

pdziugas committed:
PR Changes:
- carousel controls are displayed in the DOM
- reducing DOM operations and storing final HTML and passing it into innerHTML
- changed up function naming
- installing dev-dependencies (http and json servers) into project level instead of keeping them global to increase project portability
